realtime/plugins/azure_tts.py

```diff
@@ -93,24 +93,32 @@ def __init__(
     def viseme_received_cb(self, evt: speechsdk.SessionEventArgs):
         if (
             len(self._viseme_data["mouthCues"]) > 0
-            and (evt.audio_offset) / 10000000.0 >= self._viseme_data["mouthCues"][-1]["end"]
+            and (evt.audio_offset) / MICROSECONDS_PER_SECOND >= self._viseme_data["mouthCues"][-1]["end"]
         ):
-            self._viseme_data["mouthCues"][-1]["end"] = (evt.audio_offset) / 10000000.0
+            self._viseme_data["mouthCues"][-1]["end"] = (evt.audio_offset) / MICROSECONDS_PER_SECOND
+            # TODO: deprecate value key. keeping it for now for backwards compatibility
             self._viseme_data["mouthCues"].append(
                 {
                     "value": viseme_id_to_mouth_shapes[evt.viseme_id],
-                    "start": (evt.audio_offset) / 10000000.0 if len(self._viseme_data["mouthCues"]) > 0 else 0.0,
-                    "end": (evt.audio_offset + 1000000.0) / 10000000.0,
+                    "azure_viseme_id": evt.viseme_id,
+                    "start": (evt.audio_offset) / MICROSECONDS_PER_SECOND
```

Comment:
since visemes typically last no more than 50-100ms, we should be operating in 10s or 100s of milliseconds, instead of operating in SECONDS

Reply:
we use 1 second to be conservative. we update the viseme end when we receive a new viseme